torch/csrc/jit/tensorexpr/kernel.cpp

```diff
@@ -1859,23 +1857,23 @@ void TensorExprKernel::recompile() {
 }
 
 TensorExprKernel::TensorExprKernel(
-    const std::shared_ptr<Graph>& subgraph,
-    const std::string& kernel_func_name,
+    std::shared_ptr<Graph> subgraph,
+    std::string kernel_func_name,
     std::unordered_map<c10::Symbol, NNCLoweringFunction> custom_lowerings,
     std::vector<int64_t> symbolic_shape_inputs,
     bool pre_alloc /*= false*/,
     std::unordered_map<
         const torch::jit::Value*,
         std::vector<torch::jit::StrideInput>> symbolic_strides)
-    : graph_(subgraph),
-      code_(subgraph, ""),
+    : graph_(std::move(subgraph)),
+      code_(graph_, ""),
```

Comment:
This is a bit dangerous as members are initialized in the order they are declared in the class, not the order they are listed in here so it could cause unexpected behavior if the class is refactored. I would revert the changes to subgraph to reduce maintenance burden

Reply:
Understand your concern, I'm fine with reverting.